packages/frontend/src/pages/NewInstructLabSession.spec.ts

```diff
@@ -0,0 +1,82 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+import { vi, test, expect, beforeEach } from 'vitest';
+import { fireEvent, render } from '@testing-library/svelte';
+import type { ModelInfo } from '@shared/src/models/IModelInfo';
+import NewInstructLabSession from '/@/pages/NewInstructLabSession.svelte';
+
+const mocks = vi.hoisted(() => {
+  return {
+    // models store
+    modelsInfoSubscribeMock: vi.fn<() => ModelInfo[]>(),
+    modelsInfoQueriesMock: {
+      subscribe: (f: (msg: unknown) => void) => {
+        f(mocks.modelsInfoSubscribeMock());
+        return (): void => {};
+      },
+    },
+  };
+});
```

Comment:
why do we need the hoisted pattern there ?

Reply:
We do not use the svelte writters directly, we use a custom mechanism to be able to sync properties from the backend in the stores

https://github.com/containers/podman-desktop-extension-ai-lab/blob/8a1b04135c2629243dc2a7e1ae0936be2eaa6a3a/packages/frontend/src/stores/modelsInfo.ts#L24

We need the mocks pattern because we cannot spied the subscribe function. This is the same we are doing in all other forms mocking this store.

Reply:
AFAIK It's possible to mock, so I don't see why you're saying we can't

no need to use any hoisted

```ts
vi.mock('../stores/modelsInfo', async () => ({
  modelsInfo: {
    subscribe: vi.fn(),
    unsubscribe: vi.fn(),
  },
  }));

beforeEach(() => {
  const infos: Writable<ModelInfo[]> = writable([]);
  vi.mocked(modelsInfo).subscribe.mockImplementation((run) => (infos.subscribe(run)));
});
```

Reply:
Oh yeah that's better, thanks for noticing, I get lost with the many different manner we using vitest